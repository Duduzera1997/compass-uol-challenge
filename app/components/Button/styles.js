import styled from 'styled-components/native';

export const Container = styled.View`
  flex: 1;
  margin-top: 10%;
  align-items: center;
`;

export const TouchableButton = styled.TouchableOpacity`
  background-color: ${({ theme }) => theme.colors.primary.red};
  border-radius: 8px;
  align-items: center;
  justify-content: center;
  width: 70%;
  height: 50px;
`;

export const Label = styled.Text`
  font-family: ${({ theme }) => theme.fonts.primary.family}Bold;
  font-size: 18px;
  color: ${({ theme }) => theme.colors.secondary.white};
`;
