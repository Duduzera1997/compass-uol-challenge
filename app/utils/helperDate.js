import { format, formatDistanceToNow as formatDistance } from 'date-fns';
import ptBR from 'date-fns/locale/pt-BR';

export function formatDateLiteral(date) {
  return format(new Date(date), "dd 'de' MMMM 'de' yyyy", {
    locale: ptBR,
  });
}

export function formatDistanceToNow(date) {
  return `${formatDistance(date, { locale: ptBR })} atr√°s`;
}

export function formatDateWithDistance(date) {
  return `${formatDateLiteral(date)} | ${formatDistanceToNow(date)}`;
}
