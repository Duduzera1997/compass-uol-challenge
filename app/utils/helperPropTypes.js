import PropTypes from 'prop-types';

export const ArrayOfArticlePropTypes = PropTypes.shape({
  last_updated: PropTypes.string,
  num_results: PropTypes.number,
  results: PropTypes.arrayOf(PropTypes.shape(ArticlePropTypes)).isRequired,
});

export const ArticlePropTypes = PropTypes.shape({
  section: PropTypes.string,
  subsection: PropTypes.string,
  title: PropTypes.string,
  abstract: PropTypes.string,
  url: PropTypes.string,
  uri: PropTypes.string,
  byline: PropTypes.string,
  item_type: PropTypes.string,
  updated_date: PropTypes.string,
  created_date: PropTypes.string,
  published_date: PropTypes.string,
  material_type_facet: PropTypes.string,
  kicker: PropTypes.string,
  des_facet: PropTypes.arrayOf(PropTypes.string),
  org_facet: PropTypes.arrayOf(PropTypes.string),
  per_facet: PropTypes.arrayOf(PropTypes.string),
  geo_facet: PropTypes.arrayOf(PropTypes.string),
  multimedia: PropTypes.arrayOf(PropTypes.shape(MultimidiaPropTypes)),
  short_url: PropTypes.string,
});

export const MultimidiaPropTypes = PropTypes.shape({
  url: PropTypes.string,
  format: PropTypes.string,
  height: PropTypes.integer,
  width: PropTypes.integer,
  type: PropTypes.string,
  subtype: PropTypes.string,
  caption: PropTypes.string,
  copyright: PropTypes.string,
});
