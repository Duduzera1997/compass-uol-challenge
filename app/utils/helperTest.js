import React from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import theme from '~/styles/theme';

export const renderWithTheme = (component) =>
  render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

export const renderWithStoreAndThemeProviders = (component, { techMock }) => {
  const mockStore = configureStore();
  const state = {
    techReducer: {
      ...techMock,
    },
  };
  const store = mockStore(state);
  return renderWithTheme(<Provider store={store}>{component}</Provider>);
};
