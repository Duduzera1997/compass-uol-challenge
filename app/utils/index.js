export * from './constants';
export * from './helperDate';
export * from './helperTest';
export * from './helperPropTypes';
