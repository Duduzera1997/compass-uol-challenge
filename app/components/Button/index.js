import React from 'react';
import PropTypes from 'prop-types';
import * as SC from './styles';

function Button({ onPress }) {
  return (
    <SC.Container>
      <SC.TouchableButton onPress={onPress} testID="component-button-press">
        <SC.Label>Visualizar Not√≠cia Completa</SC.Label>
      </SC.TouchableButton>
    </SC.Container>
  );
}

Button.propTypes = {
  onPress: PropTypes.func.isRequired,
};

export default Button;
