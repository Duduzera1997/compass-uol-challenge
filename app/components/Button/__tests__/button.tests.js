import React from 'react';
import Button from '..';
import { renderWithTheme, fireEvent } from '~/utils/helperTest';

describe('Button Test', () => {
  it('should be have text label "Visualizar Notícia Completa"', () => {
    const { getByText, toJSON } = renderWithTheme(
      <Button onPress={jest.fn()} />,
    );

    expect(getByText(/Visualizar Notícia Completa/)).toBeDefined();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should be function send by prop has been called with press event', () => {
    const onPress = jest.fn();
    const { getByTestId, toJSON } = renderWithTheme(
      <Button onPress={onPress} />,
    );
    const touchable = getByTestId('component-button-press');
    fireEvent.press(touchable);
    expect(onPress).toHaveBeenCalledTimes(1);
    expect(toJSON()).toMatchSnapshot();
  });
});
