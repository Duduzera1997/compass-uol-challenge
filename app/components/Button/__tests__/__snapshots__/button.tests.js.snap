// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Test should be function send by prop has been called with press event 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginTop": "10%",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FF5A5F",
        "borderRadius": 8,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "width": "70%",
      }
    }
    testID="component-button-press"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "PTSansNarrowBold",
            "fontSize": 18,
          },
        ]
      }
    >
      Visualizar Notícia Completa
    </Text>
  </View>
</View>
`;

exports[`Button Test should be have text label "Visualizar Notícia Completa" 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginTop": "10%",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FF5A5F",
        "borderRadius": 8,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "width": "70%",
      }
    }
    testID="component-button-press"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "PTSansNarrowBold",
            "fontSize": 18,
          },
        ]
      }
    >
      Visualizar Notícia Completa
    </Text>
  </View>
</View>
`;
